geomstats/information_geometry/normal.py

```diff
@@ -116,8 +160,420 @@ def pdf(x):
         return pdf
 
 
-class NormalMetric(PullbackDiffeoMetric):
-    """Class for the Fisher information metric on normal distributions.
+class CenteredNormalDistributions(InformationManifoldMixin, SPDMatrices):
+    """Class for the manifold of centered multivariate normal distributions.
+
+    This is the class for multivariate normal distributions with zero mean.
+    Each distribution is represented by its covariance matrix, i.e. a symmetric
+    positive-definite matrix of size :math:`sample_dim`.
+
+    Parameters
+    ----------
+    sample_dim : int
+        Dimension of the sample space of the multivariate normal distribution.
+    """
+
+    def __init__(self, sample_dim):
+        super().__init__(n=sample_dim)
+        self.sample_dim = sample_dim
+        self.metric = CenteredNormalMetric(sample_dim=sample_dim)
+
+    def sample(self, point, n_samples=1):
+        """Sample from a centered multivariate normal distribution.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., sample_dim, sample_dim]
+            Symmetric positive definite matrix representing the covariance matrix
+            of a multivariate normal distribution with zero mean.
+        n_samples : int
+            Number of points to sample with each covariance matrix in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples, sample_dim]
+            Sample from centered multivariate normal distributions.
+        """
+        geomstats.errors.check_belongs(point, self)
+        if point.ndim > 3:
+            raise NotImplementedError
+        point = gs.to_ndarray(point, to_ndim=3)
+        samples = []
+        mean = gs.zeros(self.sample_dim)
+        for cov in point:
+            samples.append(gs.array(multivariate_normal.rvs(mean, cov, size=n_samples)))
+        return samples[0] if point.shape[0] == 1 else gs.stack(samples)
+
+    def point_to_pdf(self, point):
+        """Compute pdf associated to point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., sample_dim, sample_dim]
+            Symmetric positive definite matrix representing the covariance matrix
+            of a multivariate normal distribution with zero mean.
+
+        Returns
+        -------
+        pdf : function
+            Probability density function of the centered multivariate normal
+            distributions with covariance matrices provided by point.
+        """
+        geomstats.errors.check_belongs(point, self)
+        if point.ndim > 3:
+            raise NotImplementedError
+        point = point[None, :, :] if point.ndim == 2 else point
+        n = self.sample_dim
+
+        def pdf(x):
```

Comment:
following `InformationManifoldMixin`, this should be a method with signature `(x, point)`.

after this change, it is very likely that `point_to_pdf` does not need to be defined, as `InformationManifoldMixin` has a general implementation of it.

The same applies for the other manifolds.

Reply:
looking to this now, I think we should move `pdf` and `cdf` again within `point_to_pdf` and `point_to_cdf` in `InformationManifoldMixin`. (this was my suggestion @ninamiolane, but in this new code I can see what we've done is not so general as I would like.)

That's because e.g. `CenteredNormalDistributions` requires the computation of `det` and `inv` of point (i.e. it is independent of `x`), so we should do it in the scope of `point_to_pdf` instead of doing it in the `pdf` itself.

Reply:
I think this should be fixed in a different PR, so that this issue can be debated with @ninamiolane.